nav2_costmap_2d/src/costmap_layer.cpp

```diff
@@ -83,7 +83,7 @@ void CostmapLayer::updateWithTrueOverwrite(
   int max_i,
   int max_j)
 {
-  if (!enabled_) {
+  if (!enabled_ || costmap_ == nullptr) {
```

Comment:
`!enabled || !costmap_` is cleaner :)

Reply:
I understand that after the qos settings changes, this no longer gets called this is just to prevent some future issue. In that case I'd add a warning output. Also, consider that there are other update calls on the class also using `costmap_` which could also fail.

Reply:
I'd agree with you that I don think this necessarily belongs _here_ as the best indicator of a problem. If you're saying the settings fixes the issue I think we can just remove and not worry about it. This is so low in the costmap_2d package that if it were to have been an issue generally it would have been patched years ago

Reply:
Checking if a pointer is null before dereferencing it is a good defensive programming practice. I don't think it should be removed. We can add a warning if you want.

Reply:
If someone tries to use costmap with a different SLAM algorithm that uses different QOS settings than the default, without setting the parameter `map_subscribe_transient_local_` or changing the QOS settings to match, they'll hit this again. We need some type of check here. We could throw an exception instead.

Reply:
I expect this is in the callstack above the segfault. Line 328 below calls the function you modified
https://github.com/ros-planning/navigation2/blob/c1b4fe4bfecd7a3a374119b7190870112510b271/nav2_costmap_2d/plugins/static_layer.cpp#L317-L331

I believe `costmap_` should have been initialized here when the first map comes in
https://github.com/ros-planning/navigation2/blob/c1b4fe4bfecd7a3a374119b7190870112510b271/nav2_costmap_2d/plugins/static_layer.cpp#L180

but, since no map came in, resizeMaps was never called and you got the segfault.

Simply, removing the nullptr check leaves the race condition in the code. If the update loop on costmap gets called before the first map message is removed **BOOM**

If I'm right, you could 
1. add a set a flag in the second location (when the map is received) and check it in the first. IE. just bail out at the beginning like is done with enabled flag, or
2. Throw an exception where you do the nullptr check and catch it up in static layer, where you can provide a clear error message.

Reply:
@mkhansen-intel that's a good point. In light of that, I think @crdelsey option number 1 makes the most sense